### PR TITLE
feat(sbt-package): handle root relative urls

### LINKED ISSUE #13344: Renovate does not find updates of dependencies in custom maven repository
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

28.26.0

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Describe the bug

I'm trying to get updates for a Scala/SBT project. Reproduction repo is here: https://github.com/pederpansen/renovate-confluent-test

This is my `build.sbt`:

```
scalaVersion := "2.13.7"

resolvers += "Confluent Maven Repo" at "https://packages.confluent.io/maven"

libraryDependencies ++= {

  val confluentVersion = "6.0.2"

  Seq(
    "io.confluent"                % "kafka-avro-serializer"    % confluentVersion,
    "io.confluent"                % "kafka-streams-avro-serde" % confluentVersion,
  )
}
```
Renovate finds updates to all of the above dependencies except those that are located in the Confluent maven repo.

Particularly, it should find the following updates:

- kafka-avro-serializer: 6.0.2 -> 7.0.1
- kafka-streams-avro-serde: 6.0.2 -> 7.0.1

However, no updates are found. There seems to be an issue with this specific Maven repository: https://packages.confluent.io/maven/

Even explicitly adding the registry URL in `renovate.json5` does not help:

```
{
      matchDatasources: ['sbt-package'],
      registryUrls: [
        'https://repo.maven.apache.org/maven2',
        'https://packages.confluent.io/maven',
        'http://packages.confluent.io/maven',
      ],
    }
```

### Relevant debug logs

<details><summary>Logs</summary>

In the debug logs, it says the dependencies could not be found:

```
DEBUG: Looking up org.scala-lang:scala-library in repository https://repo.maven.apache.org/maven2/
DEBUG: Found 194 new releases for org.scala-lang:scala-library in repository https://repo.maven.apache.org/maven2/
DEBUG: Looking up org.scala-lang:scala-library in repository https://packages.confluent.io/maven/
DEBUG: No versions found for io.confluent:kafka-avro-serializer in 2 repositories
DEBUG: No versions found for io.confluent:kafka-streams-avro-serde in 2 repositories
DEBUG: Failed to look up dependency io.confluent:kafka-avro-serializer (io.confluent:kafka-avro-serializer)(packageFile="build.sbt", dependency="io.confluent:kafka-avro-serializer")
DEBUG: Failed to look up dependency io.confluent:kafka-streams-avro-serde (io.confluent:kafka-streams-avro-serde)(packageFile="build.sbt", dependency="io.confluent:kafka-streams-avro-serde")
DEBUG: Content is not found for Maven url
{
  "url": "https://packages.confluent.io/maven/org/scala-lang/scala-library/maven-metadata.xml"
}
DEBUG: Package releases lookups complete
{
  "baseBranch": "master"
}
DEBUG: branchifyUpgrades
DEBUG: 1 flattened updates found: scala
DEBUG: Returning 1 branch(es)
DEBUG: Fetching changelog: https://github.com/scala/scala (2.13.7 -> 2.13.8)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "config": {
    "sbt": [
      {
        "packageFile": "build.sbt",
        "deps": [
          {
            "registryUrls": [
              "https://repo.maven.apache.org/maven2",
              "https://packages.confluent.io/maven/"
            ],
            "datasource": "maven",
            "depName": "scala",
            "lookupName": "org.scala-lang:scala-library",
            "currentValue": "2.13.7",
            "separateMinorPatch": true,
            "depIndex": 0,
            "warnings": [],
            "versioning": "ivy",
            "sourceUrl": "https://github.com/scala/scala",
            "homepage": "https://www.scala-lang.org/",
            "currentVersion": "2.13.7",
            "isSingleVersion": true,
            "fixedVersion": "2.13.7",
            "updates": [
              {
                "bucket": "latest",
                "newVersion": "2.13.8",
                "newValue": "2.13.8",
                "releaseTimestamp": "2022-01-10T18:02:28.000Z",
                "newMajor": 2,
                "newMinor": 13,
                "updateType": "patch",
                "branchName": "renovate/scala-2.13.x"
              }
            ]
          },
          {
            "registryUrls": [
              "https://repo.maven.apache.org/maven2",
              "https://packages.confluent.io/maven/"
            ],
            "depName": "io.confluent:kafka-avro-serializer",
            "lookupName": "io.confluent:kafka-avro-serializer",
            "currentValue": "6.0.2",
            "groupName": "confluentVersion for io.confluent",
            "datasource": "sbt-package",
            "depIndex": 1,
            "warnings": [
              {
                "topic": "io.confluent:kafka-avro-serializer",
                "message": "Failed to look up dependency io.confluent:kafka-avro-serializer"
              }
            ],
            "versioning": "ivy",
            "updates": []
          },
          {
            "registryUrls": [
              "https://repo.maven.apache.org/maven2",
              "https://packages.confluent.io/maven/"
            ],
            "depName": "io.confluent:kafka-streams-avro-serde",
            "lookupName": "io.confluent:kafka-streams-avro-serde",
            "currentValue": "6.0.2",
            "groupName": "confluentVersion for io.confluent",
            "datasource": "sbt-package",
            "depIndex": 2,
            "warnings": [
              {
                "topic": "io.confluent:kafka-streams-avro-serde",
                "message": "Failed to look up dependency io.confluent:kafka-streams-avro-serde"
              }
            ],
            "versioning": "ivy",
            "updates": []
          }
        ]
      }
    ]
  }
}
```

In the http statistics, I can see the Confluent Maven repo URL was accessed once:

```
http statistics
{
  "urls": {
    "https://api.github.com/graphql (POST)": 4,
    "https://api.github.com/repos/pederpansen/renovate-confluent-test/contents/default.json (GET)": 1,
    "https://api.github.com/repos/pederpansen/renovate-confluent-test/contents/renovate.json5 (GET)": 1,
    "https://api.github.com/repos/pederpansen/renovate-confluent-test/issues/1 (GET)": 2,
    "https://api.github.com/repos/pederpansen/renovate-confluent-test/pulls (GET)": 1,
    "https://api.github.com/repos/whitesource/merge-confidence/contents/beta.json (GET)": 1,
    "https://packages.confluent.io/maven/io/confluent/ (GET)": 1,
    "https://repo.maven.apache.org/maven2/org/scala-lang/scala-library/2.13.8/scala-library-2.13.8.pom (GET)": 1
  },
  "hostStats": {
    "api.github.com": {
      "requestCount": 10,
      "requestAvgMs": 212,
      "queueAvgMs": 0
    },
    "packages.confluent.io": {
      "requestCount": 1,
      "requestAvgMs": 88,
      "queueAvgMs": 0
    },
    "repo.maven.apache.org": {
      "requestCount": 1,
      "requestAvgMs": 34,
      "queueAvgMs": 0
    }
  },
  "totalRequests": 12
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by JonasTebben**
## Changes:

Replaces root relative URLs from sbt-package repositories with relative URLs which makes it possible to fetch releases from Confluent sbt-package repository (issue #13344).

## Context:

Closes #13344

## Documentation (please check one with an [x])

[x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Newly added/modified unit tests

### CODE REVIEW

File: lib/datasource/sbt-plugin/util.ts [hunk 1]
```diff
@@ -1,12 +1,25 @@
+import url from 'url';
```
Comment (viceice) [inline lib/datasource/sbt-plugin/util.ts:1]: Nope, use our url util or global URL object.

Comment (JonasHavers) [reply lib/datasource/sbt-plugin/util.ts:1]: The import is done in `lib/datasource/maven/util.ts` like this, but no problem, I'll also change it there.

Comment (viceice) [reply lib/datasource/sbt-plugin/util.ts:1]: https://github.com/renovatebot/renovate/pull/9109 😉

Comment (JonasHavers) [reply lib/datasource/sbt-plugin/util.ts]: Done :)

Comment (viceice) [reply lib/datasource/sbt-plugin/util.ts]: https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting 😉

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs deconflicting

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**